Runtime/ObjectMap.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using static NodeApi.JSNativeApi;
+using static NodeApi.JSNativeApi.Interop;
+
+namespace NodeApi;
+
+/// <summary>
+/// Tracks JS constructors and instance JS wrappers for exported classes, enabling
+/// .NET objects to be automatically wrapped when returned to JS, and re-wrapped as needed
+/// if the (weakly-referenced) JS wrapper has been released.
+/// </summary>
+internal static class ObjectMap
```

Comment:
Instead of `static`, this class needs to support multiple instances so that each instantiation of a module can track objects separately. That's necessary to resolve the `multi_instance` test failure.